### DISCUSSION TITLE
Should the tar_load RStudio addin work?

### DISCUSSION THREAD

**Discussion by kendonB** [ropensci/targets#974] [Help]
When I run the addin, I see this:
```
Loading target mytarget into global environment.
```
Then the target prints but is not loaded into the environment. I can see in the code it is running `tar_load` (and not `tar_read`) so I'm guessing the environment isn't set correctly somehow. 

These don't look like they're in the targets manual, so I was wondering if they're a work in progress.

**Comment by wlandau** [comment]
If you also see "✖ Identified no targets to load.", then the target may not exist in the data store yet. Otherwise, it is hard to say why it doesn't work on your end, I just checked again on my end and the target loads fine. The code behind the addin is in `R/rstudio_addin_tar_load.R`, and it does give `tar_load()` the global environment:

https://github.com/ropensci/targets/blob/32ef529c1c9819c4f1676a67efe28c476e71be6a/R/rstudio_addin_tar_load.R#L17

**Comment by kendonB** [comment]
Hi Will,

I tried again using the latest `targets-minimal`.

Ran the macro and got this:
```
• Loading target raw_data into global environment.
# A tibble: 153 × 6
   Ozone Solar.R  Wind  Temp Month   Day
   <dbl>   <dbl> <dbl> <dbl> <dbl> <dbl>
 1    41     190   7.4    67     5     1
 2    36     118   8      72     5     2
 3    12     149  12.6    74     5     3
 4    18     313  11.5    62     5     4
 5    NA      NA  14.3    56     5     5
 6    28      NA  14.9    66     5     6
 7    23     299   8.6    65     5     7
 8    19      99  13.8    59     5     8
 9     8      19  20.1    61     5     9
10    NA     194   8.6    69     5    10
# … with 143 more rows
# ℹ Use `print(n = ...)` to see more rows
> raw_data
Error: object 'raw_data' not found
```
I'm on Rstudio 2022.07.2 Build 576 and Ubuntu 22.04.

**Reply by wlandau** [reply]
Sorry, still can't reproduce the problem. Could depend on how you are using the addin. The addin assumes your cursor is on the name of the target where it is typed in a text editor window.

**Comment by kendonB** [comment]
How's this for strange - it works when running via point and click but not via keyboard shortcut:

First is point and click (Tools -> Addins) and second is keyboard shortcut
```
• Loading target raw_data into global environment.
> raw_data
# A tibble: 153 × 6
   Ozone Solar.R  Wind  Temp Month   Day
   <dbl>   <dbl> <dbl> <dbl> <dbl> <dbl>
 1    41     190   7.4    67     5     1
 2    36     118   8      72     5     2
 3    12     149  12.6    74     5     3
 4    18     313  11.5    62     5     4
 5    NA      NA  14.3    56     5     5
 6    28      NA  14.9    66     5     6
 7    23     299   8.6    65     5     7
 8    19      99  13.8    59     5     8
 9     8      19  20.1    61     5     9
10    NA     194   8.6    69     5    10
# … with 143 more rows
# ℹ Use `print(n = ...)` to see more rows
> rm(raw_data)
• Loading target raw_data into global environment.
# A tibble: 153 × 6
   Ozone Solar.R  Wind  Temp Month   Day
   <dbl>   <dbl> <dbl> <dbl> <dbl> <dbl>
 1    41     190   7.4    67     5     1
 2    36     118   8      72     5     2
 3    12     149  12.6    74     5     3
 4    18     313  11.5    62     5     4
 5    NA      NA  14.3    56     5     5
 6    28      NA  14.9    66     5     6
 7    23     299   8.6    65     5     7
 8    19      99  13.8    59     5     8
 9     8      19  20.1    61     5     9
10    NA     194   8.6    69     5    10
# … with 143 more rows
# ℹ Use `print(n = ...)` to see more rows
> raw_data
Error: object 'raw_data' not found
```

**Reply by wlandau** [reply]
That's odd. I tried bonding CTRL+SHIFT+t to the `tar_load()` addin, and it worked that way too. Maybe your shortcut conflicts with other addins or commands, and maybe this problem shows up with other addins.

The code for the addin is in `R/rstudio_addin_tar_load.R` in an exported function called `rstudio_addin_tar_load()`. You could try `debug(rstudio_addin_tar_load)` and then use the interactive debugger to check e.g. what is different about the `context` argument depending on how you invoke it.
